File: C/Iniciante/1118variasnotascomvalidacao.c
#include <stdio.h>
int main() {
    int i=0,tipo;
    double n,soma=0;
    do {
        do {
            scanf("%lf", &n);
            if (n >= 0 && n <= 10) {
                soma += n;
                i++;
            } else {
                printf("nota invalida\n");
            }
        } while (i < 2);
        printf("media = %.2lf\n",soma/2);
        i=0;
        soma=0;
        while (1) {
            printf("novo calculo (1-sim 2-nao)\n");
            scanf("%d", &tipo);
            if(tipo==1 || tipo==2)break;
        }
    }while(tipo!=2);
    return 0;
}

File: Python/iniciante/1118variasnotascomvalidacao.py
soma = 0
tipo = 0
while tipo != 2:
    i = 0
    while i < 2:
        n = int(input())
        if 0 <= n <= 10:
            soma += n
            i += 1
        else:
            print('nota invalida')
    print("media = %.2lf\n" % soma / 2)
    i=0
    soma=0
    while True:
        print("novo calculo (1-sim 2-nao)")
        tipo=int(input())
        if tipo==1 or tipo==2:
            break
